segmentation/utils/synthetic_spatial_datagen.py

```diff
@@ -222,10 +221,163 @@ def generate_test_label_map(size_img=(1024, 1024), num_A=100, num_B=100, num_C=1
+        size_img (tuple): the dimensions of the image we wish to generate
+        nuc_radius (int): the radius of the nucleus of each cell
+
+    Returns:
+        sample_nuclear_signal (numpy): an array of equal dimensions to segmentation_mask
+            which have nuclear signal generated for the provided cell ids
+    """
+
+    # define the nuclear signal array
+    sample_nuclear_signal = np.zeros(segmentation_mask.shape)
+
+    # default to cell 1 if cell_ids not set
+    if not cell_ids:
+        cell_ids = [1]
+
+    for cell in cell_ids:
+        center = cell_centers[cell]
+
+        # generate the nuclear region in the middle of the cell with the same cell center
+        # and set signal to a uniform value
+        nuc_region_x, nuc_region_y = circle(center[0], center[1], nuc_radius, shape=size_img)
+        sample_nuclear_signal[nuc_region_x, nuc_region_y] = 1
+
+        # let's keep things simple for now and not include jitter or anything
+        # that can easily be included in the next commit
+
+    return sample_nuclear_signal
+
+
+def generate_two_cell_test_membrane_signal(segmentation_mask, cell_centers, cell_ids=None,
+                                           size_img=(1024, 1024), cell_radius=10, memb_diameter=5):
+    """
+    This function generates membrane signal for the provided cells
+
+    Args:
+        segmentation_mask (numpy): an array which contains the labeled cell regions
+        cell_centers (dict): a dictionary which contains the centers associated with each cell region
+        cell_ids (list): a list of cells we wish to generate nuclear signal for, if None assume just cell 2
+        size_img (tuple): the dimensions of the image we wish to generate
+        cell_radius (int): the radius of the entire cell, needed to do proper circle subtraction
+            for a ring-shaped membrane
+        memb_diameter (int): the diameter of the membrane ring of each cell
+
+    Returns:
+        sample_membrane_signal (numpy): an array of equal dimensions to segmentation_mask
+            which have membrane signal generated for the provided cell ids
+    """
+
+    # define the nuclear signal array
+    sample_membrane_signal = np.zeros(segmentation_mask.shape)
+
+    # default to cell 2 if cell_ids not set
+    if not cell_ids:
+        cell_ids = [2]
+
+    for cell in cell_ids:
+        center = cell_centers[cell]
+
+        # generate both the coordinates of the cell region and non-membrane region
+        # for proper circle subtraction to generate membrane
+        cell_region_x, cell_region_y = circle(center[0], center[1], cell_radius, shape=size_img)
+        non_memb_region_x, non_memb_region_y = circle(center[0], center[1], cell_radius, shape=size_img)
+
+        # perform circle subtraction
+        sample_membrane_signal[cell_region_x, cell_region_y] = 1
+        sample_membrane_signal[non_memb_region_x, non_memb_region_y] = 1
```

Comment:
shouldn't the id numbers (1 and 1) be different here (1 and 0(2?)) or am I missing something?

Reply:
Crap, yes it should be 1 and 0.